synapseclient/client.py

```diff
@@ -980,7 +980,10 @@ def store(self, obj, **kwargs):
             # update the file_handle metadata if the FileEntity's FileHandle id has changed
             if '_file_handle' in local_state \
                     and properties['dataFileHandleId'] != local_state['_file_handle'].get('id', None):
-                local_state['_file_handle'] = self._getFileHandle(properties['dataFileHandleId'])
+                local_state['_file_handle'] = find_data_file_handle(
+                    self._getEntityBundle(properties['id'], bitFlags=0x800 | 0x1)
```

Comment:
Nice, the reason both the filehandle and entity bigflag are specified is because `find_data_file_handle` requires both.

Reply:
I think a better fix would to be also to switch over to https://rest-docs.synapse.org/rest/POST/entity/id/bundle2.html for entity bundles as well so that we can avoid using bitflags. Now that i have more time to work on the client I'll do that before checking this in